packages/gatsby-plugin-typescript/src/gatsby-node.js

```diff
@@ -23,16 +22,10 @@ module.exports.modifyWebpackConfig = (
   // error (i.e., not build) at something or other.
   const opts = { compilerOptions: copts, transpileOnly }
 
-  // Load gatsby babel plugin to extract graphql query
-  const extractQueryPlugin = path.resolve(
-    __dirname,
-    `../gatsby/dist/utils/babel-plugin-extract-graphql.js`
-  )
-
   config.loader(`typescript`, {
     test,
     loaders: [
-      `babel?${JSON.stringify({ plugins: [extractQueryPlugin] })}`,
+      `babel?${JSON.stringify(babelConfig)}`,
```

Comment:
rather then passing through the babel config can you stick the ts loader in front of the existing js loader in the chain. You can also "override" the `js` loader and pass a function the gets the current one and alters it.

its not clean, but this will be much easier to do neatly in v2, and i don't think we want to add more surface area if we can avoid it, since its much harder to deprecate later

Reply:
If I do that then typescript becomes a hard dependency. I don't think that makes sense for most people using gatsby.